Dawidw furachanges

DawidGit opened the pull request.

DawidGit committed:
tworzenie klasy order

dfura committed:
add registration features

dfura committed:
reimplementation registration and auth services

dfura committed:
add registration features

dfura committed:
reimplementation registration and auth services

dfura committed:
Merge remote-tracking branch 'origin/spring-security-implementation' into spring-security-implementation

# Conflicts:
#	src/main/java/pl/connectis/electronicswebshop/web/login/EncryptPass.java
#	src/main/java/pl/connectis/electronicswebshop/web/login/SecSecurityConfig.java
#	src/main/java/pl/connectis/electronicswebshop/web/login/UserRepository.java

dfura committed:
add roles/privileges
add thymeleaf
add custom templates
cleanup code

dfura committed:
add custom UserDetailsService
add security GrantedAuthorities support
restructure project

dfura committed:
add registration mvc

DawidGit committed:
Merge remote-tracking branch 'origin/spring-security-implementation' into dawidw

# Conflicts:
#	src/main/java/pl/connectis/electronicswebshop/web/login/EncryptPass.java
#	src/main/java/pl/connectis/electronicswebshop/web/login/SecSecurityConfig.java
#	src/main/java/pl/connectis/electronicswebshop/web/login/UserRepository.java
#	src/main/java/pl/connectis/electronicswebshop/web/registration/UserRegistrationController.java
#	src/main/java/pl/connectis/electronicswebshop/web/security/SecurityConfig.java
#	src/main/resources/application.properties

DawidGit committed:
dodanie produkt i Order

dfura committed:
dodanie wyswietlania listy produktow
przygotowanie funkcjonalnosci dod. prod. do zamowienia

dfura committed:
add products to order adding functionality
cleanup and refactor

DawidGit committed:
zrobiony basket

dfura committed:
modify rearranged order/product code
add new initial data

DawidGit committed:
wrzucone quantity do basketa i dodanie delete

dfura committed:
rename ProductQuantity -> OrderLine
add article deletion method
fix security spring: role -> SimpleGrantedAuthority

DawidGit committed:
poprawienie importowania product ID z htmla

dfura committed:
rebase

dfura committed:
add employee registration feature

dfura committed:
add allordersview products listing

dfura committed:
modify allorderview + product quantities view

dfura committed:
add page navigation, thymeleaf security integration

dfura committed:
add bootstrap css

dfura committed:
add header view

dfura committed:
fix header overlap

dfura committed:
add header dynamics

DawidGit committed:
dodana validacja do formularza dodawania artykułu

dfura committed:
cleanup and reformat

dfura committed:
small cleanup
fix deleteProductefromorder

dfura committed:
remove garbage

dfura committed:
cleanup

mateusz-dziurdziak commented on pom.xml:
```diff
@@ -19,19 +19,28 @@
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-data-jpa</artifactId>
+        <dependency>
```
Masz duży bałagan w zależnościach. Nie wszystkie są wymagane, a część się powtarza np. dodając `spring-boot-starter-web` nie musisz już dodawać `spring-webmvc` oraz `spring-web`

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/InitialDataLoader.java:
```diff
@@ -0,0 +1,127 @@
+    }
+
+    @Transactional
+    Privilege createPrivilegeIfNotFound(String name) {
```
@Transactional nie działa jeśli jest zaaplikowany na metodach, które nie są publiczne. Co więcej jako, że metoda jest wywoływana wyłącznie z innych metod tej klasy powinna być prywatna. 

Póki metoda `onApplicationEvent` się nie zakończy wszystko i tak dzieje się w transakcji

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/InitialDataLoader.java:
```diff
@@ -0,0 +1,127 @@
+        return privilege;
+    }
+
+    @Transactional
```
Jak wyżej - metoda powinna być prywatna + Transactional jest niepotrzebny

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/InitialDataLoader.java:
```diff
@@ -0,0 +1,127 @@
+        Privilege privilege = privilegeRepository.findByName(name);
+        if (privilege == null) {
+            privilege = new Privilege(name);
+            privilegeRepository.save(privilege);
```
Dobrym zwyczajem jest korzystanie z tego co zwróci metoda `save` czyli:
```
privilege = privilageRepository.save(privilege);
```
Warstwa dostępu do bazy danych może w jakiś sposób zmodyfikować wstawiane dane więc najlepiej wykorzystać "nowy" obiekt

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/InitialDataLoader.java:
```diff
@@ -0,0 +1,127 @@
+import java.util.*;
+
+@Component
+public class InitialDataLoader implements ApplicationListener<ContextRefreshedEvent> {
```
`ContextRefreshedEvent` moze zostać wzbudzony wielokrotnie w czasie działania aplikacji. Zamiast tego wykorzystaj sposób inicjalizacji implementując `ApplicationRunner`. 
Idealnie taka inicjalizacja powinna również działać wyłącznie czasami (nie chcemy by uruchamiała się za każdym razem oraz np w środowisku produkcyjnym). Można to osiągnąć wykorzystując adnotacje @Profile lub @ConditionalOnProperty

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/Order.java:
```diff
@@ -0,0 +1,85 @@
+
+@Entity
+@Table(name = "orders")
+public class Order {
```
W projekcie znajduje się biblioteka `lombok` więc nie ma potrzeby pisania getterów i setterów

DawidGit replied:
poprawione

DawidGit replied:
gotowe

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/Order.java:
```diff
@@ -0,0 +1,85 @@
+    private final LocalDate orderDate = LocalDate.now();
+
+    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
+    public final Collection<OrderLine> products = new HashSet<>();
```
pola powinny być prywatne

DawidGit replied:
poprawione

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderRepository.java:
```diff
@@ -0,0 +1,20 @@
+
+    Order findByAddedByAndOrderStatus(String addedBy, OrderStatus orderStatus);
+
+    Order findByAddedBy(String username);
```
3/5 methods tutaj nie jest użytych w kodzie

mateusz-dziurdziak replied:
I równocześnie metody `Order find*` w przypadku uruchomienia rzuciłyby wyjątek jeśli jest więcej niż jeden `Order` spełniający kryterium wyszukiwania (np. po added by)

DawidGit replied:
wyrzucone

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderService.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    @Override
+    public Order addOrder() {
```
Zamiast pisać cała metodę w środku można po prostu zawołać `return addOrder("anonymous")`

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderService.java:
```diff
@@ -0,0 +1,119 @@
+        return orderRepository.findById(lastOpenOrder.getId()).get().getOrderLines();
+    }
+
+    public void deleteOrder(Principal principal, OrderStatus orderStatus) {
```
parametr `orderStatus` nie jest używany

DawidGit replied:
usunięte

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderService.java:
```diff
@@ -0,0 +1,119 @@
+            if (products.getProduct().equals(product)) {
+                orderLine = products;
+                orderLine.setQuantity(orderLine.getQuantity() + quantity);
+                product.setStock(product.getStock() - quantity);
```
Linia
` product.setStock(product.getStock() - quantity);`
jest niepotrzebnie powtórzona. Mogłaby być napisana wyłącznie raz

DawidGit replied:
zrobione

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderService.java:
```diff
@@ -0,0 +1,119 @@
+        return true;
+    }
+
+    public Order removeLine(long orderId, long productId, int quantity) {
```
Nazewnictwo tej metody mogłoby być lepsze - usuwasz tutaj produk z korzyka

DawidGit replied:
nazwa zmieniona na removeProduct

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderService.java:
```diff
@@ -0,0 +1,119 @@
+    public Order removeLine(long orderId, long productId, int quantity) {
+        Optional<Order> foundOrder = orderRepository.findById(orderId);
+        if (foundOrder.isPresent()) {
+            for (Iterator<OrderLine> iterator = foundOrder.get().getOrderLines().iterator(); iterator.hasNext(); ) {
```
Użyłeś iteratora by potem usunąć dany produkt ale chyba czytelniej by było jakbyś rozbił to najpierw na znalezienie a potem na operację

```
OrderLine orderLine = null;
for (OrderLine currentOrderLine : orderLines) {
    if (currentOrderLine.getProduct().getId().equals productId) {
        orderLine = currentOrderLine; 
        break;
     }
}
// zrob resztę tutaj
```

DawidGit replied:
gotowe

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderService.java:
```diff
@@ -0,0 +1,119 @@
+                Product product = orderLine.getProduct();
+                if (orderLine.getProduct().getId().equals(productId)) {
+                    if (orderLine.getQuantity() - quantity <= 0) {
+                        product.setStock(orderLine.getQuantity() + quantity);
```
Nie ma tutaj błędu logicznego? Stock nie powinien być `product.getStock() + orderLine.getQuatity()`?

DawidGit replied:
poprawione

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/OrderService.java:
```diff
@@ -0,0 +1,119 @@
+                        iterator.remove();
+                        orderLineRepository.delete(orderLine);
+                    } else {
+                        product.setStock(orderLine.getQuantity() + quantity);
```
Podobnie jak wyżej. Wygląda na bład w logice

DawidGit replied:
też poprawione

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/products/ProductService.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    public Product getProductByID(Long productID) {
+        if (productsRepository.findById(productID).isPresent())
```
`return productsRepository.findById(productID).orElse(null)`

DawidGit replied:
zrobione

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/products/ProductsController.java:
```diff
@@ -0,0 +1,105 @@
+            Model model,
+            Principal principal
+    ) {
+        if (quantity < 0) return "error1";
```
Formatowanie oraz powinieneś zawsze używać `{}` przy `if`

DawidGit replied:
poprawione

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/service/OrderServiceInterface.java:
```diff
@@ -0,0 +1,13 @@
+
+import java.util.Optional;
+
+public interface OrderServiceInterface {
```
Obecnie przyjmuje się, że tworzenie interface w przypadku gdy tylko jedna klasa je implementuje jest niepotrzebne

DawidGit replied:
myślimy o rozwoju aplikacji więc zostawimy

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/service/OrderServiceInterface.java:
```diff
@@ -0,0 +1,13 @@
+public interface OrderServiceInterface {
+    Order addOrder();
+
+    Optional<Order> findById(Long id);
```
Metoda nieużywana

DawidGit replied:
metoda usunięta

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/service/ProductServiceInterface.java:
```diff
@@ -0,0 +1,11 @@
+
+public interface ProductServiceInterface {
+
+    void addProduct(Product product);
```
Metoda nieużywana

DawidGit replied:
wywalone

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/web/auth/MyUserDetailsService.java:
```diff
@@ -0,0 +1,67 @@
+
+        List<String> privileges = new ArrayList<>();
+        List<Privilege> collection = new ArrayList<>();
+        for (Role role : roles) {
```
Nie ma potrzeby rozbijania tego na dwie pętle. Z tego co widzę wystarczyłaby jedna po `roles`

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/products/ProductDto.java:
```diff
@@ -0,0 +1,47 @@
+import javax.validation.constraints.NotNull;
+import java.security.Principal;
+
+@Component
```
Czy to musi być `Component`?

DawidGit replied:
adnotacja usunięta

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/web/registration/UserDto.java:
```diff
@@ -0,0 +1,82 @@
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+@PasswordMatches
```
Walidacja hasła (oraz jakakolwiek inna) nie działają dla tego obiektu. Próba dodania użytkownika bez niczego kończy się błędem typu 500.

Może gdzieś brakuje adnotacji @Valid lub @Validated ?

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/web/security/PasswordConfig.java:
```diff
@@ -0,0 +1,15 @@
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder(10);
```
Używając `BCrypPasswordEncoder` dane, które znajdują się w bazie danych podlegają odtworzeniu. Zamiast odwracalnych algorytmów powinno się używać funkcji mieszających (hash) poprzez np. `MessageDigestPasswordEncoder`

Nachri replied:
Żródła twierdzą, że jest to hashujący algorytm i ma również wbudowane solenie.
https://www.baeldung.com/spring-security-registration-password-encoding-bcrypt
https://en.wikipedia.org/wiki/Bcrypt

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/web/validation/EmailValidator.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EmailValidator
```
Podobnie jak z hasłem ten walidator również nie jest uruchamiany

mateusz-dziurdziak reviewed (changes requested):
Ogólnie sporo rzeczy do poprawy. Skup się głównie na:
- poprawie formatowania w całym projekcie (jest do tego gotowa akcja w ide)
- wyczyść kod z warningów generowanych przez ide 
- sprawdź czy wszystko co jest zawarte w kodzie rzeczywiście jest wykorzystywane (jak np walidatory)
- pousuwaj maksymalnie niepotrzebne gettery i settery (od tego masz lombok)

Nie przechodziłem przez pliki *.html skupiając się raczej na warstwie backend.

W projekcie brakuje testów.

DawidGit committed:
first corrections Dawid W

dfura committed:
refactor InitialDataLoader
add Spring profiles dev,test

dfura committed:
remove boilerplate code with Lombok

dfura committed:
remove unused dependencies

dfura committed:
add dummy Order for test,dev

dfura committed:
fix missing dependencies
fix validations for registration views
add enums representation for roles,privileges

dfura committed:
css styling for registration page
code refactor

dfura committed:
fix null basketCount @ /login

dfura committed:
change access modifiers for ProductDto
cleanup ProductDto class
return saved Entities for ProductService

dfura committed:
add ProductService, ProductRepository tests
logic strength fix for basketCount

DawidGit committed:
cleanUp code (removed useless annotations and method's parameters)
added tests
corrected logic in methods

DawidGit committed:
Merge remote-tracking branch 'origin/dawidw-furachanges' into dawidw-furachanges

# Conflicts:
#	src/main/java/pl/connectis/electronicswebshop/InitialDataLoader.java
#	src/main/java/pl/connectis/electronicswebshop/order/Order.java
#	src/main/java/pl/connectis/electronicswebshop/products/ProductService.java
#	src/main/java/pl/connectis/electronicswebshop/service/ProductServiceInterface.java
#	src/main/resources/application.properties

dfura committed:
cleanup and ordercontrollertest fix

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/order/Order.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
W przypadku JPA equals i hashCode powinny zależeć wyłącznie od pól oznaczonych jako Id

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/persistence/model/User.java:
```diff
@@ -0,0 +1,98 @@
+
+@Entity
+@Table(name="users")
+public class User {
```
lombok

mateusz-dziurdziak commented on src/main/java/pl/connectis/electronicswebshop/products/ProductService.java:
```diff
@@ -0,0 +1,58 @@
+        return productsRepository.findAll();
+    }
+
+    public List<Product> getAllProductsByOrderId(Order order) {
```
Przefiltowane produkty powinny być już zwracane z repozytorium. Obecnie iterujecie przez wszystkie produkty i wszystkie zamówienia by to otrzymać

dfura committed:
refactor equals and hashcode with lombok

dfura committed:
remove unused method

dfura committed:
fix addProduct NullPointer error
fix /login redirect
